client/consensus/src/tests.rs

```diff
@@ -716,3 +290,590 @@ async fn authorities_runtime_api_tests() {
+    // the cancellation token should stop it before though
+    let relay_client = RelayChain {
+        client: client.clone(),
+        block_import_iterations: 1000u32,
+    };
+    let spawner = sp_core::testing::TaskExecutor::new();
+    let orchestrator_tx_pool = sc_transaction_pool::BasicPool::new_full(
+        Default::default(),
+        true.into(),
+        None,
+        spawner.clone(),
+        client.clone(),
+    );
+    let mock_runtime_api = MockRuntimeApi::new(Some(1000u32.into()));
+
+    let (overseer, handle) = dummy_overseer_builder(spawner.clone(), MockSupportsParachains, None)
+        .unwrap()
+        .replace_runtime_api(|_| mock_runtime_api)
+        .build()
+        .unwrap();
+    spawner.spawn("overseer", None, overseer.run().then(|_| async {}).boxed());
+
+    let cancellation_token = CancellationToken::new();
+
+    // Build the collator
+    let params = LookAheadParams {
+        create_inherent_data_providers: move |_block_hash, _| async move {
+            let slot = InherentDataProvider::from_timestamp_and_slot_duration(
+                Timestamp::current(),
+                SlotDuration::from_millis(SLOT_DURATION_MS),
+            );
+
+            Ok((slot,))
+        },
+        block_import: environ.0.clone(),
+        relay_client: relay_client.clone(),
+        keystore: keystore.into(),
+        para_id: 1000.into(),
+        proposer: ConsensusProposer::new(environ.clone()),
+        collator_service: CollatorService::new(
+            client.clone(),
+            Arc::new(spawner.clone()),
+            Arc::new(move |_, _| {}),
+            Arc::new(environ.clone()),
+        ),
+        authoring_duration: Duration::from_millis(500),
+        cancellation_token: cancellation_token.clone(),
+        code_hash_provider: DummyCodeHashProvider,
+        collator_key: CollatorPair::generate().0,
+        force_authoring: false,
+        get_orchestrator_aux_data: move |_block_hash, _extra| async move {
+            let aux_data = OrchestratorAuraWorkerAuxData {
+                authorities: vec![alice_public.into()],
+                // This is the orchestrator consensus, it does not have a slot frequency
+                slot_freq: None,
+            };
+
+            Ok(aux_data)
+        },
+        get_current_slot_duration: move |_block_hash| SlotDuration::from_millis(6_000),
+        overseer_handle: OverseerHandle::new(handle),
+        relay_chain_slot_duration: Duration::from_secs(6),
+        para_client: environ.clone().into(),
+        sync_oracle: DummyOracle,
+        para_backend: backend,
+        orchestrator_client: environ.into(),
+        orchestrator_slot_duration: SlotDuration::from_millis(SLOT_DURATION_MS),
+        orchestrator_tx_pool,
+    };
+
+    // we issue the cancellation token, which should exist the loop after creating the first blocks
+    tokio::spawn(async move { cancellation_token.cancel() });
```

Comment:
`.cancel()` is not async

```suggestion
    cancellation_token.cancel();
```

Reply:
fixed in [82242eb](https://github.com/moondance-labs/tanssi/pull/662/commits/82242eb5310be2e51fad9bb311e231f63c8199bc)